### PR TITLE
feat: Move creation of 'ext' HNS network into cse for Windows

### LINKED ISSUE #2432: Move 'ext' network creation into CSE provisioning for Windows nodes
**Describe the request**
Currently an 'ext' network is created once on each Windows node running AzureCNI in the script that starts running the kubelet. This should be configured in the CSE and not in the script that starts the kubelet.

**Explain why AKS Engine needs it**
Failure to create the 'ext' network should manifest as a deployment failure. Currently it is confusing to customers to have a "successful" aks-engine / AKS node pool deployment but not have the nodes joined to the cluster because the kubelet never started.

This is also a supportability issue

**Describe the solution you'd like**
Move logic to create the 'ext' network into the CSE

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

**Issue Fixed**:

Fixes #2432

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 1]
```diff
@@ -275,12 +279,14 @@ try
                     -IdentitySystem "{{ GetIdentitySystem }}"
             }
 
-        } elseif ($global:NetworkPlugin -eq "kubenet") {
+            New-ExternalHnsNetwork
```
Comment (PatrickLang) [inline parts/k8s/kuberneteswindowssetup.ps1]: just to clarify - is `New-ExternalHnsNetwork` not required for kubenet?

Comment (marosset) [reply parts/k8s/kuberneteswindowssetup.ps1]: It isn't being created for kubenet today

Comment (marosset) [reply parts/k8s/kuberneteswindowssetup.ps1]: So i assume it isn't

Comment (marosset) [reply parts/k8s/kuberneteswindowssetup.ps1]: The removal is in the code block for if ($NetworkPlugin -eq "azure") { }

File: parts/k8s/windowsazurecnifunc.ps1 [hunk 2]
```diff
@@ -255,3 +255,38 @@ function GenerateAzureStackCNIConfig
...
+    Write-Log "Creating new HNS network `"ext`""
+    $externalNetwork = "ext"
+    $na = @(Get-NetAdapter -Physical)
+
+    if ($na.Count -eq 0) {
+        throw "Failed to find any physical network adapters"
+    }
+
+    # If there is more than one adapter, use the first adapter.
+    $managementIP = (Get-NetIPAddress -ifIndex $na[0].ifIndex -AddressFamily IPv4).IPAddress
```
Comment (PatrickLang) [inline parts/k8s/windowsazurecnifunc.ps1]: @pradipd - should this be smarter? @dineshgovindasamy said there may be cases where a NIC doesn't have an IP and we should choose another?

File: parts/k8s/kuberneteswindowssetup.ps1 (hunk 1 above)
Comment (marosset) [reply parts/k8s/kuberneteswindowssetup.ps1]: Maybe @pradipd  or @dineshgovindasamy  can comment on what scenarios we should create the ext HNS network?

File: parts/k8s/windowsazurecnifunc.ps1 (hunk 2 above)
Comment (PatrickLang) [reply parts/k8s/windowsazurecnifunc.ps1]: This is re: https://github.com/Azure/aks-engine/issues/2281

Comment (pradipd) [reply parts/k8s/windowsazurecnifunc.ps1]: Yes. Actually, the (Get-NetAdapter -Physical) on L261 needs to be different for acceleratedNetworkingEnabledWindows.
I just haven't had a chance to figure out what it should be.

Comment (pradipd) [reply parts/k8s/windowsazurecnifunc.ps1]: Yes. Actually, the (Get-NetAdapter -Physical) on L261 needs to be different for acceleratedNetworkingEnabledWindows.
I just haven't had a chance to figure out what it should be.

File: parts/k8s/kuberneteswindowssetup.ps1 (hunk 1 above)
Comment (dineshgovindasamy) [reply parts/k8s/kuberneteswindowssetup.ps1]: It should be created if you want to prevent that brief network disconnect on reboot

Comment (marosset) [reply parts/k8s/kuberneteswindowssetup.ps1]: I updated to create the network for all network plugings.

File: parts/k8s/windowsazurecnifunc.ps1 (hunk 2 above)
Comment (PatrickLang) [reply parts/k8s/windowsazurecnifunc.ps1]: @pradipd ok - we'll consider that for a separate PR

**Pull request merged**